edi-tools/modules/x12xsd/x12xsd.bal

```diff
@@ -169,4 +182,13 @@ function getUnitElement(string name, xml x12xsd) returns xml|error {
         return error("EDI segment/segment group not found in the input schema. Unit name: " + name);
     }
     return unitElement;
+}
+
+public function getBalCompatibleName(string rawName) returns string {
+    string name = rawName.trim();
+    name = regex:replaceAll(name, "[^a-zA-Z1-9_]", "_");
```

Comment:
use the `regexp` langlib is better because the `regex` library is deprecated..

Reply:
Will update this in the next verison.